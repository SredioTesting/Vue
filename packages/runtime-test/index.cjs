'use strict'

if (process.env.NODE_ENV === 'production') {
  module.exports = require('./dist/runtime-test.prod.cjs')
} else {
  module.exports = require('./dist/runtime-test.cjs')
}
