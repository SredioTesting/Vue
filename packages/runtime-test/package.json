{
  "name": "@vue/runtime-test",
  "version": "3.2.21",
  "description": "@vue/runtime-test",
  "private": true,
  "main": "index.cjs",
  "module": "dist/runtime-test.esm-bundler.mjs",
  "types": "dist/runtime-test.d.ts",
  "files": [
    "index.cjs",
    "dist"
  ],
  "repository": {
    "type": "git",
    "url": "git+https://github.com/vuejs/vue-next.git",
    "directory": "packages/runtime-test"
  },
  "keywords": [
    "vue"
  ],
  "author": "Evan You",
  "license": "MIT",
  "bugs": {
    "url": "https://github.com/vuejs/vue-next/issues"
  },
  "homepage": "https://github.com/vuejs/vue-next/tree/master/packages/runtime-test#readme",
  "dependencies": {
    "@vue/shared": "3.2.21",
    "@vue/runtime-core": "3.2.21"
  }
}
