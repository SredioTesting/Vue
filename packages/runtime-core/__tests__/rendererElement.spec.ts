import { nodeOps, render, h, serialize, Text, VNode } from '@vue/runtime-test'

describe('renderer: element', () => {
  test('simple', () => {
    const root = nodeOps.createElement('div')

    render(h('div'), root)

    expect(serialize(root)).toMatchSnapshot()
  })

  test('with props', () => {
    const root = nodeOps.createElement('div')
    let id = 'test1'

    render(
      h('div', {
        id
      }),
      root
    )
    expect(serialize(root)).toMatchSnapshot()

    id = 'test2'

    render(
      h('div', {
        id
      }),
      root
    )
    expect(serialize(root)).toMatchSnapshot()
  })

  test('with direct text children', () => {
    const root = nodeOps.createElement('div')
    let text = 'Hello, world!'

    render(h('div', text), root)
    expect(serialize(root)).toMatchSnapshot()

    text = 'I ❤ Vue'

    render(h('div', text), root)
    expect(serialize(root)).toMatchSnapshot()
  })

  test('with text node children', () => {
    const root = nodeOps.createElement('div')

    render(h('div', [h(Text, ['I ❤ Vue'])]), root)

    expect(serialize(root)).toMatchSnapshot()
  })

  test('handle already mounted VNode', () => {
    const root = nodeOps.createElement('div')
    const vnode = h('test')
    const tree1 = h('div', vnode)
    const tree2 = h('div', vnode)

    render(tree1, root)
    expect(serialize(root)).toMatchSnapshot()
    expect((tree1.children as VNode[])[0]).toMatchObject(vnode)

    render(tree2, root)
    expect(serialize(root)).toMatchSnapshot()
    expect((tree2.children as VNode[])[0]).toMatchObject(vnode)
  })
})
