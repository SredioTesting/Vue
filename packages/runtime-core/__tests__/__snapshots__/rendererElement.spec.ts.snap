// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderer: element handle already mounted VNode: root 1 1`] = `"<div><div></div></div>"`;

exports[`renderer: element handle already mounted VNode: root 2 1`] = `"<div><div></div></div>"`;

exports[`renderer: element simple 1`] = `"<div><div></div></div>"`;

exports[`renderer: element with direct text children 1`] = `"<div><div>I ❤ Vue</div></div>"`;

exports[`renderer: element with props 1`] = `"<div><div id=\\"test\\"></div></div>"`;

exports[`renderer: element with text node children 1`] = `"<div><div>I ❤ Vue</div></div>"`;
